design/authn_azure/authn_azure_solution_design.md

```diff
@@ -0,0 +1,508 @@
+| System-assigned identity               | an identity for an instance in AD that is trusted. The identity is tied to the resource and abides to its lifecycle. In other words, if the resource is delete so to will the identity                                                                        |
+| User-assigned identity                 | an identity for an instance in AD that is trusted. This identity can be assigned to 1+ Azure services instances and is independent of instance lifecycle                                                                                                      |
+| Azure Instance Metadata Service (IMDS) | a REST endpoint accessible to all VMs created via the Azure Resource Manager. Resources in Azure request access tokens from the Azure Instance Metadata service endpoint. We will first focused on supporting VMs and explore other services at a later time  |
+| JWT                                    | JSON-based access token that asserts claims (logged in, etc)                                                                                                                                                                                                  |
+| xms_mirid                              | a claim that defines the resource that holds the encoding of the instance's subscription, resource group, and name needed for authorization                                                                                                                   |
+
+## Useful links
+
+- [Feature doc](https://app.zenhub.com/workspaces/appliance-development-5c9cf04bfa1c204b63eddcc6/issues/cyberark/conjur/1266)
+
+## Issue description
+
+Many of our customers are using Azure as their main cloud provider. Therefore we want to enable Azure resources to authenticate according to Azure 
+properties and get back a Conjur access token in order to retrieve secrets from our providers. At current, customers can authenticate their Azure 
+VMs with Conjur by creating a Conjur host identity for their Azure resources. The Azure resource would then receive an API key as part of the 
+creation of a Conjur identity which they must use to connect to targets. This is not optimal nor convenient because we are not allowing the 
+resources to authenticate based on their Azure identity and demand that the resource hold an extra secret, the Conjur-given API key.
+
+## Solution
+
+We will add a new authenticator to Conjur, which users can authenticate with from Azure VMs. As mentioned in the [feature doc](https://app.zenhub.com/workspaces/appliance-development-5c9cf04bfa1c204b63eddcc6/issues/cyberark/conjur/1266), before any authentication 
+request is sent to Conjur, the admin will load the authenticator policy: 
+
+```yaml
+# policy id needs to match the convention `conjur/authn-azure/<service ID>`
+- !policy
+  id: conjur/authn-azure/prod
+  body:
+  - !webservice
+    annotations: 
+      provider-uri: https://sts.windows.net/TENANT_ID
```

Comment:
@sigalsax i would change this to 
```
provider-uri: <Azure AD URI> # For example: https://sts.windows.net/df242c82-fe4a-47e0-b0f4-e3cb7f8104f2/
```

Reply:
Changing it back to variable as per solution design meeting